test/test_base.py

```diff
@@ -98,6 +98,7 @@ def test_object_resolution(self):
     def test_with_bare_rw_repo(self, bare_rw_repo):
         assert bare_rw_repo.config_reader("repository").getboolean("core", "bare")
         assert osp.isfile(osp.join(bare_rw_repo.git_dir, "HEAD"))
+        assert osp.isdir(bare_rw_repo.working_dir)
```

Comment:
Thanks! 

Could you also add an assertion of the `working_tree_dir` for completeness? It's easy to confuse both which is what I did apparently. The `working_dir` is the CWD of the git repository, either the `working_tree_dir` or the `git_dir`, which I think should be an assertion for the bare and non-bare cases.

If that assertion holds, of course, I was going by the docs.

<img width="701" alt="Screenshot 2023-02-02 at 07 15 07" src="https://user-images.githubusercontent.com/63622/216246481-c70d0788-e210-4b8e-a003-3042f30b5034.png">

Reply:
Ok so the docs in `base.py` say that working_tree_dir can return none
```python
    @property
    def working_tree_dir(self) -> Optional[PathLike]:
        """:return: The working tree directory of our git repository. If this is a bare repository, None is returned."""
        return self._working_tree_dir
```

I've added a few more assertions, please check if I understood you.

Reply:
Also updated the python docs of the class to mention returning None